### PR TITLE
Ignore MaskedArrayFutureWarning in table

### LINKED ISSUE #6251: DeprecationWarning from _get_delta_tdb_tt in astropy.time
With Python 3.6 and astropy master (`3.0.dev19204`), I'm now getting an `AstropyDeprecationWarning` and a `MaskedArrayFutureWarning` when doing a simple AltAz to RaDec coordinate conversion that used to work without warnings:
```
WARNING: AstropyDeprecationWarning: The longitude property is deprecated and may be removed in a future version.
        Use `lon` instead. [astropy.time.core]
/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/table/column.py:1233: MaskedArrayFutureWarning: setting an item on a masked array which has a shared mask will not copy the mask and also change the original mask array in the future.
Check the NumPy 1.11 release notes for more information.
  ma.MaskedArray.__setitem__(self, index, value)
```

To reproduce:
```python
import astropy.units as u
from astropy.coordinates import EarthLocation, AltAz, SkyCoord
from astropy.time import Time

# We want to see the traceback where the warnings come from:
# https://stackoverflow.com/a/22376126/498873
import traceback
import warnings
import sys
def warn_with_traceback(message, category, filename, lineno, file=None, line=None):
    log = file if hasattr(file,'write') else sys.stderr
    traceback.print_stack(file=log)
    log.write(warnings.formatwarning(message, category, filename, lineno, line))
warnings.showwarning = warn_with_traceback

# Now the code that creates a DeprecationWarning in Astropy master
location = EarthLocation(lat=48.8567 * u.deg, lon=2.3508 * u.deg)
obstime = Time.now()
altaz = AltAz(obstime=obstime, location=location)
coord = SkyCoord('5:34:31.97', '22:0:52.10', unit=(u.hour, u.deg), frame='icrs')
crab_altaz = coord.transform_to(altaz)
```

Output with tracebacks showing where the warnings come from:
```
$ python altaz_example.py 
  File "altaz_example.py", line 21, in <module>
    crab_altaz = coord.transform_to(altaz)
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/coordinates/sky_coordinate.py", line 461, in transform_to
    new_coord = trans(self.frame, generic_frame)
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/coordinates/transformations.py", line 1121, in __call__
    curr_coord = t(curr_coord, curr_toframe)
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/coordinates/transformations.py", line 745, in __call__
    res = self.func(fromcoord, toframe)
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/coordinates/builtin_frames/icrs_cirs_transforms.py", line 30, in icrs_to_cirs
    jd1, jd2 = get_jd12(cirs_frame.obstime, 'tdb')
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/coordinates/builtin_frames/utils.py", line 113, in get_jd12
    newtime = getattr(time, scale)
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/time/core.py", line 1093, in __getattr__
    tm._set_scale(attr)
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/time/core.py", line 465, in _set_scale
    args.append(get_dt(jd1, jd2))
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/time/core.py", line 1271, in _get_delta_tdb_tt
    lon = location.longitude
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/utils/decorators.py", line 119, in deprecated_func
    warnings.warn(message, category, stacklevel=2)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/warnings.py", line 99, in _showwarnmsg
    msg.file, msg.line)
  File "altaz_example.py", line 12, in warn_with_traceback
    traceback.print_stack(file=log)
/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/time/core.py:1271: AstropyDeprecationWarning: The longitude property is deprecated and may be removed in a future version.
        Use `lon` instead.
  lon = location.longitude
  File "altaz_example.py", line 21, in <module>
    crab_altaz = coord.transform_to(altaz)
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/coordinates/sky_coordinate.py", line 461, in transform_to
    new_coord = trans(self.frame, generic_frame)
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/coordinates/transformations.py", line 1121, in __call__
    curr_coord = t(curr_coord, curr_toframe)
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/coordinates/transformations.py", line 745, in __call__
    res = self.func(fromcoord, toframe)
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/coordinates/builtin_frames/cirs_observed_transforms.py", line 53, in cirs_to_altaz
    xp, yp = get_polar_motion(obstime)
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/coordinates/builtin_frames/utils.py", line 43, in get_polar_motion
    xp, yp, status = iers.IERS_Auto.open().pm_xy(time, return_status=True)
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/utils/iers/iers.py", line 585, in open
    cls.iers_table = cls.read(file=filename)
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/utils/iers/iers.py", line 458, in read
    table = cls._combine_a_b_columns(iers_a)
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/utils/iers/iers.py", line 390, in _combine_a_b_columns
    table = cls._substitute_iers_b(table)
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/utils/iers/iers.py", line 705, in _substitute_iers_b
    table['UT1_UTC_B'][:n_iers_b] = iers_b['UT1_UTC'].value
  File "/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/table/column.py", line 1233, in __setitem__
    ma.MaskedArray.__setitem__(self, index, value)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/numpy/ma/core.py", line 3395, in __setitem__
    MaskedArrayFutureWarning, stacklevel=2)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/warnings.py", line 99, in _showwarnmsg
    msg.file, msg.line)
  File "altaz_example.py", line 12, in warn_with_traceback
    traceback.print_stack(file=log)
/Users/deil/Library/Python/3.6/lib/python/site-packages/astropy/table/column.py:1233: MaskedArrayFutureWarning: setting an item on a masked array which has a shared mask will not copy the mask and also change the original mask array in the future.
Check the NumPy 1.11 release notes for more information.
  ma.MaskedArray.__setitem__(self, index, value)
```
I didn't check if it's the same in the 2.0 branch, but probably it is and would be good to fix before the release, no?

cc @taldcroft @mhvk

### PR DESCRIPTION
**Pull request by pllim**
Fix #6251 . Also address a warning seen in #6140.

@mhvk, is this what you meant in https://github.com/astropy/astropy/issues/6251#issuecomment-310201726 ?

Travis is running on my fork -- https://travis-ci.org/pllim/astropy/builds/247239902 (I cancelled the one here due to long queue.)

### CODE REVIEW

**Comment by saimn** [comment]
See also #5060.

**Comment by pllim** [comment]
@saimn , from https://github.com/astropy/astropy/issues/5060#issuecomment-264178578 and also what @mhvk has been saying, it seems like this is the right thing to do?

**Comment by saimn** [comment]
@pllim - Yes, looks good :+1:

**Comment by mhvk** [comment]
Just to be sure, is this a warning users will actually see, or just something that comes up during testing? If the latter, it is probably best to capture/turn it off only during testing. (Sorry, am in the middle of a conference still...)

cc @taldcroft

**Comment by pllim** [comment]
@cdeil reported it as well in #6251 but only he can comment it bothers him or not. As far as testing goes, this warning does **not** cause failure in Sphinx 1.6+ although would be nice if we can get rid of it (but again, not required).

**Comment by saimn** [comment]
Yes it's possible to see this warning as a user:
```python
In [2]: t = Table([np.ma.masked_invalid([1,np.nan,3])], masked=True)

In [3]: t['col0'].mask
Out[3]: array([False,  True, False], dtype=bool)

In [4]: t['col0'][1] = 1
/home/simon/lib/astropy/astropy/table/column.py:1233: MaskedArrayFutureWarning: setting an item on a masked array which has a shared mask will not copy the mask and also change the original mask array in the future.
Check the NumPy 1.11 release notes for more information.
  ma.MaskedArray.__setitem__(self, index, value)
```
It's probably better to filter this warning as users can do nothing about it (and probably do not understand what it means).

**Comment by bsipocz** [comment]
@taldcroft @mhvk - Any decisions on this? Can we merge?

**Comment by mhvk** [comment]
I think this is fine, though perhaps better to just make it numpy version dependent - that way it is easier to remove once it becomes obsolete. @taldcroft - could you have a look?

File: astropy/table/column.py [hunk 1]
```diff
@@ -1230,7 +1237,9 @@ def __setitem__(self, index, value):
 
         # update indices
         self.info.adjust_indices(index, value, len(self))
-        ma.MaskedArray.__setitem__(self, index, value)
+        with warnings.catch_warnings():
+            warnings.simplefilter('ignore', MaskedArrayFutureWarning)
```
Comment (taldcroft) [inline astropy/table/column.py]: With the alias of `MaskedArrayFutureWarning = FutureWarning` for the case of no `MaskedArrayFutureWarning`, could this code hide a valid `FutureWarning`?  Or is that not possible?

**Comment by eteq** [comment]
@pllim - (FYI, if you want to address @taldcroft's suggestion, can you try to do it before the end of today? If not we'll probably just merge and backport so that it can get in the RC tonight, seeing as how @taldcroft approved it)

File: astropy/table/column.py (hunk 1 above)
Comment (pllim) [inline astropy/table/column.py]: Theoretically, definitely possible. I can put `if ... else ...` here instead but not sure how much of a performance hit that would be, but maybe I'm pre-maturely attempting to optimize this.

Comment (taldcroft) [inline astropy/table/column.py]: The `if .. else ..` will be negligible compared to other stuff in the setitem.  So I would suggest to make the change and put in a `HAS_MASKED_ARRAY_FUTURE_WARNING` variable in the import try/except above.

**Comment by pllim** [comment]
@taldcroft , @eteq -- I changed the logic; Do you like it better this way?

**Comment by pllim** [comment]
See https://travis-ci.org/pllim/astropy/builds/247239902 for Travis test on my fork.

File: astropy/table/column.py (hunk 1 above)
Comment (pllim) [inline astropy/table/column.py]: I just did before I saw your reply. `if ... else ...` is there, except I just use the warning variable itself as a switch; Is that acceptable to you?

**Comment by bsipocz** [comment]
Passing CI, so I'm merging.

**Pull request merged**